### PR TITLE
Correct typo in servlet api dependabot exclusion

### PR DESCRIPTION
**Pull request by timja**
Noticed when reviewing https://github.com/jenkinsci/jenkins/pull/6588

it's correct on this line:
https://github.com/jenkinsci/jenkins/blob/229299fb9cb72be3b15693c6a8fc02e2bbdb850a/.github/dependabot.yml#L31-L32

### CODE REVIEW

**Review by basil — COMMENTED** [review]
Lines 30 and 31 prevent updates of `javax.servlet:javax.servlet-api` in https://github.com/jenkinsci/jenkins/blob/229299fb9cb72be3b15693c6a8fc02e2bbdb850a/core/pom.xml#L470-L475= and the reason is because `javax.servlet:javax.servlet-api` is provided by Jetty. So the version of `javax.servlet:javax.servlet-api` we depend on in `provided` scope should be the same as that of Jetty (currently Servlet API 3.x for Jetty 9.4.x).

Lines 22 and 23 (being deleted in this PR) are intended to prevent updates of `javax.servlet:servlet-api` (note the different artifact ID!) in https://github.com/jenkinsci/jenkins/blob/master/pom.xml#L115-L132= which itself is a trick added in commit 982a8faf5a apparently to prevent plugins (and core?) from bundling Servlet API 2.x. `javax.servlet:servlet-api` has been banned through Enforcer in the plugin parent POM since jenkinsci/plugin-pom@a6089678898e84d427d2ded0e34f3e7371b6dc2e and in the core parent POM since jenkinsci/pom@82f73fef, so I think it may no longer be necessary to keep this trick in core, and keeping it complicates `jenkinsci/jenkins/pom.xml` and our Dependabot configuration, so it may well be best to dispense with it. However, if we are to keep it, then we need to ensure Dependabot doesn't update it, and the way to ensure that is to add a Dependabot exclusion for `javax.servlet:servlet-api` (distinct from the Dependabot exclusion for `javax.servlet:javax.servlet-api`).

In short, my suggestion is to maintain logical consistency by taking one of two actions:

1. Removing `javax.servlet:servlet-api` from both the core dependency tree and the Dependendabot exclusions list, relying on the abovementioned Enforcer bans to keep it out of our dependency trees.
2. Retain the existing trick and update the Dependabot exclusion to correctly refer to `javax.servlet:servlet-api` on lines 22 and 23.

The current PR does one half of the first option but not the other half, exposing us to unwanted updates to this dependency. I have no strong preference between these two options; the former is a bit more aggressive but likely fine, and the second is the more conservative option.

**Review by basil — APPROVED** [review]
